content/docs/latest/reference/developer-guides/build-tips-and-tricks.md

```diff
@@ -0,0 +1,100 @@
+
+#### Ideal Future Situation
+
+This document outlines the best ways to achieve the fastest builds possible in terms of hardware, software, and workflow. The Flatcar team is also considering instrumenting the build process to provide more detailed telemetry, using tools like `otel-cli`.
+
+#### Hardware Requirements
+
+- **CPU**: AMD64 box with a minimum of 4-8 real cores. Increasing cores beyond 32 offers only marginal speedups.
+- **Storage**: NVMe storage is a must, ideally PCIe 3. Benchmark with:
+  ```bash
+  dd if=/dev/random of=test.img oflag=direct,sync bs=4k count=81920 status=progress
+  ```
+  Target a result of around 20-30MB/s.
+
+- **RAM**: Use RAMFS/TMPFS for optimal performance. Ensure at least 64 GB of RAM, as the build itself requires 6-8 GB, and TMPFS needs 30-50 GB.
+  ```bash
+  dd if=/dev/random of=test.img oflag=direct,sync bs=4k count=81920 status=progress
+  ```
+
+  Target 200-300MB/s using RAMDISK.
+
+- **Disk Space**: Ensure at least 25-50 GB of free space depending on the number of builds.
+
+- **Network**: Fast Internet is essential, as Flatcar's dockerized environment requires downloading the 10GB SDK image. During the build, several Rust crates and Go modules are downloaded from source repositories.
+
+#### Software Requirements
+
+- **Docker Engine**: If using RAMDISK, ensure Docker’s root-data is set to the RAMDISK mount point.
+- **qemu**: Required for ARM64 builds and for testing VM images.
+
+#### Workflow for Building Flatcar
+
+1. **Create RAMDISK**: For the fastest build times, set up a RAMDISK for storage.
+2. **Install Docker Engine**: Set the Docker root-data to the RAMDISK mount point if using RAMDISK.
+3. **Clone the Flatcar Scripts**:
+   ```bash
+   git clone https://github.com/flatcar/scripts
+   ```
+   Clone the repository onto the RAMDISK for best speedups.
+4. **Run the SDK Container**:
+   ```bash
+   ./run_sdk_container -t
+   ```
+   This command downloads the latest Docker SDK container image, which is around 10GB.
+5. **Build Packages**:
+   ```bash
+   ./build_packages
+   ```
+6. **Build the Image**:
+   ```bash
+   ./build_image
+   ```
+7. **Convert Image for VM Testing**:
+   ```bash
+   ./image_to_vm.sh <params>
+   ```
+   Use `qemu-kvm` to create a VM instance and test the image.
+
+#### Additional Tips and Tricks
+
+- **Binary Packages**: Use `-U` with `./run_sdk_container` if you want to use binary packages, though full builds generally rebuild everything.
+- **Tmpfs**: Using Tmpfs can provide additional speedups. However, in some cases, using NVMe might not differ significantly.
+- **Parallel Fetch**: Enable `parallel-fetch` in Portage to download files in the background during the build.
+- **Rust/Go Module Download Bottleneck**: A significant bottleneck is the sequential download of Rust crates and Go modules. Parallel downloads or HTTP connection reuse could help speed this up.
```

Comment:
ditto `future dev work` - remove `Parallel downloads or HTTP connection reuse could help speed this up. Comment `